src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -827,7 +827,16 @@ public Solution WithProjectParseOptions(ProjectId projectId, ParseOptions option
                 return this;
             }
 
-            return this.ForkProject(newProject, CompilationTranslationAction.ProjectParseOptions(newProject));
+            if (this.Workspace.PartialSemanticsEnabled)
+            {
+                // don't fork tracker with queued action since access via partial semantics can become inconsistent (throw).
+                // Since changing options is rare event, it is okay to start compilation building from scratch.
+                return this.ForkProject(newProject, forkTracker: false);
```

Comment:
When we drop the tracker we also lose the dependencies on other compilations that may get GC'd before we get back to reconstructing it.  So it may cause additional unnecessary work to occur.